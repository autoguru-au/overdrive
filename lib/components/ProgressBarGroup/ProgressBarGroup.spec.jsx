import { render } from '@testing-library/react';
import * as React from 'react';

import { ProgressBarGroup } from '.';

describe('<ProgressBarGroup />', () => {
	it('should not throw', () => {
		expect(() =>
			render(<ProgressBarGroup values={[1, 2, 3]} />),
		).not.toThrow();
	});

	it('should match the snapshot', () => {
		expect(
			render(<ProgressBarGroup values={[1, 2, 3]} />).container
				.firstChild,
		).toMatchSnapshot();
	});

	it('should should render a prefix for every value', () => {
		expect(
			render(
				<ProgressBarGroup
					values={[1, 2, 3]}
					prefixLabels={['a', 'b', 'c']}
				/>,
			).container.firstChild,
		).toMatchSnapshot();
	});

	it('should should ignore extra labels', () => {
		expect(
			render(
				<ProgressBarGroup
					values={[1, 2, 3]}
					prefixLabels={['a', 'b', 'c', 'd']}
				/>,
			).container.firstChild,
		).toMatchSnapshot();
	});

	it('should should ignore not break when not enough labels', () => {
		expect(
			render(<ProgressBarGroup values={[1, 2, 3]} prefixLabels={['a']} />)
				.container.firstChild,
		).toMatchSnapshot();
	});
});
