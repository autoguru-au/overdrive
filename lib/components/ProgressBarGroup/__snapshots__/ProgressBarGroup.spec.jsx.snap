// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ProgressBarGroup /> > should match the snapshot 1`] = `
<div
  class="theme_OD_Base__j49br80"
  data-od-component-provider=""
>
  <div
    class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_alignItems_center_mobile__onxwjujr ProgressBarGroup_root__1e1wroi0"
    style="grid-template-columns: 1fr;"
  >
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    />
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 16.666666666666664%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    />
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 33.33333333333333%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    />
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 50%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
  </div>
</div>
`;

exports[`<ProgressBarGroup /> > should should ignore extra labels 1`] = `
<div
  class="theme_OD_Base__j49br80"
  data-od-component-provider=""
>
  <div
    class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_alignItems_center_mobile__onxwjujr ProgressBarGroup_root__1e1wroi0"
    style="grid-template-columns: auto 1fr;"
  >
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    >
      a
    </span>
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 16.666666666666664%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    >
      b
    </span>
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 33.33333333333333%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    >
      c
    </span>
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 50%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
  </div>
</div>
`;

exports[`<ProgressBarGroup /> > should should ignore not break when not enough labels 1`] = `
<div
  class="theme_OD_Base__j49br80"
  data-od-component-provider=""
>
  <div
    class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_alignItems_center_mobile__onxwjujr ProgressBarGroup_root__1e1wroi0"
    style="grid-template-columns: auto 1fr;"
  >
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    >
      a
    </span>
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 16.666666666666664%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    />
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 33.33333333333333%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    />
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 50%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
  </div>
</div>
`;

exports[`<ProgressBarGroup /> > should should render a prefix for every value 1`] = `
<div
  class="theme_OD_Base__j49br80"
  data-od-component-provider=""
>
  <div
    class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_alignItems_center_mobile__onxwjujr ProgressBarGroup_root__1e1wroi0"
    style="grid-template-columns: auto 1fr;"
  >
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    >
      a
    </span>
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 16.666666666666664%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    >
      b
    </span>
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 33.33333333333333%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330 ProgressBarGroup_label__1e1wroi1"
    >
      c
    </span>
    <div
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_gray100__onxwju3d ProgressBar_container__4k1c010"
    >
      <div
        class="reset_resetVariants__4z7xt3b reset_resetVariants_as_div__4z7xt3c reset_block__4z7xt33 sprinkles_borderRadius_1_mobile__onxwju9j sprinkles_backgroundColor_green500__onxwju3q ProgressBar_bar__4k1c011 ProgressBar_container__4k1c010"
        style="width: 50%;"
      />
    </div>
    <span
      class="reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  reset_resetVariants__4z7xt3b reset_resetVariants_as_span__4z7xt3k  sprinkles_fontSize_3__onxwju20 sprinkles_lineHeight_3__onxwju29 sprinkles_fontWeight_normal__onxwju2g sprinkles_color_light__onxwju6l typography_typographyStyles__yhjp330"
    />
  </div>
</div>
`;
